tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/AsynchronousCSTest.java

```diff
@@ -78,20 +79,9 @@ public static WebArchive deploy() {
      */
     @Test
     public void testAsyncIsNotFinished() {
-        CompletableFuture<Void> waitingFuture = newWaitingFuture();
-        
-        CompletionStage<Connection> resultFuture = client.serviceCS(waitingFuture);
-
-        Assert.assertFalse(completesQuickly(resultFuture));
-
-        complete(waitingFuture);
-
-        try {
-            waitUntilCompleted(resultFuture);
-        }
-        catch (CompletionException e) {
-            handleCompletionException(e);
-        }
+        Task taskToPerform = new ServiceTask();
+        CompletionStage<Task> taskResult = client.serviceCS(taskToPerform);
+        assertFalse(taskResult.toCompletableFuture().isDone());
```

Comment:
This `assertFalse(taskResult.isDone())` check is not sufficient to prove correctness.

`serviceCS` is an asynchronous method, so `isDone()` could always be false, even if the implementation is not correct.

The old check ensures that `isDone()` continues to return `false` for 500ms to give the task the opportunity to complete if it's going to.

Reply:
Annoyingly, I can't find a way to have awaitility do this sort of check